mlir/utils/performance/tuningRunner.py

```diff
@@ -59,42 +60,48 @@ def verifyKernelWithPerfConfig(perfConfig, config, paths: Paths, options: Option
     profilerCommand = [perfRunner.ROCPROF, '--stats', paths.mlir_paths.cpu_runner_path] + mlirCpuRunnerArgs
 
     if options.debug:
-        print(rocmlirGenCommand)
-
-    # invoke rocmlir-gen.
-    p1 = subprocess.Popen(rocmlirGenCommand.split(), stdout=subprocess.PIPE, stderr=subprocess.DEVNULL)
-    # pipe to rocmlir-driver
-    p2 = subprocess.Popen(rocmlirDriverCommand, stdin=p1.stdout, stdout=subprocess.PIPE, stderr=subprocess.DEVNULL)
-    p1.stdout.close() # Allow p1 to receive a SIGPIPE if p2 exits.
-    # pipe to rocprof + mlir-cpu-runner.
-    p3 = subprocess.Popen(profilerCommand, stdin=p2.stdout, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-    p2.stdout.close() # Allow p2 to receive a SIGPIPE if p3 exits.
-    # get output.
-    try:
-        outs, errs = p3.communicate(timeout=600)
-        outs = outs.decode('utf-8')
-        if len(errs) > 0 or not CORRECT_RESULT_RE.search(outs):
-            print(f"""Verification failed:
+        print(rocmlirGenCommand, file=sys.stderr)
+
+    prevdir = os.getcwd()
+    with tempfile.TemporaryDirectory() as tmpdir:
```

Comment:
Make a temp directory, cd to it, execute using the usual filename, extract the nanoSeconds.  The 'with' removes the directory, the 'finally' ensures that we always cd back where we came from.

Reply:
Makes sense - I assume this auto-deletes after?